crates/database/src/byte_array.rs

```diff
@@ -0,0 +1,84 @@
+use sqlx::{
+    encode::IsNull,
+    error::BoxDynError,
+    postgres::{PgArgumentBuffer, PgHasArrayType, PgTypeInfo, PgValueFormat, PgValueRef},
+    Decode, Encode, Postgres, Type,
+};
+
+/// Wrapper type for fixed size byte arrays compatible with sqlx's Postgres implementation.
+pub struct ByteArray<const N: usize>(pub [u8; N]);
+
+impl<const N: usize> Type<Postgres> for ByteArray<N> {
+    fn type_info() -> PgTypeInfo {
+        <[u8] as Type<Postgres>>::type_info()
+    }
+}
+
+impl<const N: usize> PgHasArrayType for ByteArray<N> {
+    fn array_type_info() -> PgTypeInfo {
+        <[&[u8]] as Type<Postgres>>::type_info()
+    }
+}
+
+impl<const N: usize> Decode<'_, Postgres> for ByteArray<N> {
+    fn decode(value: PgValueRef<'_>) -> Result<Self, BoxDynError> {
+        let mut bytes = [0u8; N];
+        match value.format() {
+            // prepared query
+            PgValueFormat::Binary => {
+                bytes = value.as_bytes()?.try_into().unwrap();
```

Comment:
Is `unwrap` intentional here? Shouldn't this be an error if you have a binary value with the wrong number of bytes?

Reply:
That was leftover from testing. Thanks for catching. Adjusted test to make sure it errors in this case.